common/wireaddr.c

```diff
@@ -453,15 +472,69 @@ bool parse_wireaddr_internal(const char *arg, struct wireaddr_internal *addr,
 	if (strstarts(arg, "autotor:")) {
 		addr->itype = ADDR_INTERNAL_AUTOTOR;
 		return parse_wireaddr(arg + strlen("autotor:"),
-				      &addr->u.torservice, 9051,
+				      &addr->u.torservice.torservice_address, 9051,
+				      dns_ok ? NULL : &needed_dns,
+				      err_msg);
+	}
+
+	/* 'statictor:' is a special prefix meaning talk to Tor to create
+	 * an onion address. */
+	if (strstarts(arg, "statictor:")) {
```

Comment:
Let's test this once, maybe, not 4 times?

Reply:
Actually, let's use tal_strspit() here:

```
if (strstarts(arg, "statictor:)) {
    /* Format is separated by colons. */
    char **parts = tal_strsplit(tmpctx, arg, ":", STR_EMPTY_OK);
    for (size_t i = 1; i < tal_count(parts)-1; i++) { 
        if (streq(parts[i], "torport")) {
            char **endp;
            addr->u.torservice.port = strtol(parts[i], &endp, 10);
            if (addr->u.torservice.port == 0 || endp != '\0') {
                *err_msg = "Bad :torport: number";
                return false;
            }
        } else if (streq(pars[i], "torblob")) {
		...
```

Reply:
done, now with tal_strsplit, incl. ipv6 ::  handling. TODO: needs some test